libgalois/include/katana/EntityTypeManager.h

```diff
@@ -0,0 +1,227 @@
+#pragma once
+
+#include <bitset>
+#include <optional>
+#include <unordered_map>
+#include <vector>
+
+#include "katana/Logging.h"
+
+namespace katana {
+
+/// EntityTypeID uniquely identifies an entity (node or edge) type
+/// EntityTypeID for nodes is distinct from EntityTypeID for edges
+/// This type may either be an atomic type or an intersection of atomic types
+/// EntityTypeID is represented using 8 bits
+using EntityTypeID = uint8_t;
```

Comment:
Could this be an OpaqueID? I know it may not be worth the effort, but I think it's worth considering doing it now.

Reply:
I think the end goal may be to have the width of this type change dynamically based on how many types are in a given graph. If that is no longer the case then I agree that we should make these opaque.